### PR TITLE
feat: #64 kotlin language linter support

### PR DESCRIPTION
**Pull request by isaac-heist-slalom**
- Adds support for Kotlin linting.
- Updated script to support 2 or 3 digit story numbers.

Closes [#64](https://github.com/slalombuild/secureli/issues/64)

AC's confirmed:

1.  Pre-commit hooks are configured and config is stored in the repo under /.secureli/config when a Kotlin repo is detected.

image

image

image

  
2.  If there are any failing linting issues, per the list above, when a commit is attempted the commit is blocked and an error with details is displayed to the user.

image

3.  Readme & internal help is updated with supporting language info:
(See updated contributing.md file)

### CODE REVIEW

**Review by kevin-orlando — APPROVED** [review]

File: scripts/get-current-branch.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ else
 fi
 
 # Check if short name work
-if [[ $branch == "main" || $branch =~ ^secureli[0-9][0-9][0-9]$ || $branch =~ "dependabot" ]]; then
+if [[ $branch == "main" || $branch =~ ^secureli[0-9][0-9]?[0-9]?$ || $branch =~ "dependabot" ]]; then
```
Comment (tdurk93) [inline scripts/get-current-branch.sh:30]: Can we do this instead?
```suggestion
if [[ $branch == "main" || $branch =~ ^secureli[0-9]+$ || $branch =~ "dependabot" ]]; then
```

Comment (isaac-heist-slalom) [reply scripts/get-current-branch.sh]: Yes, that makes sense and is more maintainable. Thanks. Fixed!

**Review by kevin-orlando — APPROVED** [review]

**Review by tdurk93 — APPROVED** [review]
I tested this locally and it works with against test `.kt` file. My main concern is that this doesn't actually perform linting, but rather static analysis for other issues. That said, static analysis is something we are also providing for other languages (and fits in well with seCureLI's security focus), so I don't think it's a bad idea for `detekt` to be added here as a pre-commit hook.

It looks like there's another tool called [ktlint](https://github.com/pinterest/ktlint) that provides linting. I found a [pre-commit hook](https://github.com/maltzj/ktlint-pre-commit) for it, but it appears to not be maintained, and I don't love how it hard-codes the version number of `ktlint` to download.

An alternative solution is to either roll our own pre-commit hook to add `ktlint`, or I found that `ktlint` is supported as a [plugin for detekt](https://detekt.dev/docs/rules/formatting/). We would just need to figure out how best to download the plugin as part of the hook execution and how to check for updates.

I think this extra functionality should be done in a future spike ticket, so I think this PR is good to go.

**Comment by tdurk93** [comment]
I added a spike ticket for future work: https://github.com/slalombuild/secureli/issues/392

**Pull request merged**